File: EasyFinance.BusinessLogic/Models/ReceiptFilterCriteria.cs
using System;

namespace EasyFinance.BusinessLogic.Models
{
    public class ReceiptFilterCriteria
    {
        public int? UserId { get; set; }

        public DateTime? DateFrom { get; set; }

        public DateTime? DateTo { get; set; }

        public int? CategoryId { get; set; }

        public decimal? TotalFrom { get; set; }

        public decimal? TotalTo { get; set; }
    }
}

File: EasyFinanceUI/src/app/models/receipt-filter-criteria.ts
export class ReceiptFilterCriteria {
    public userId: number;
    public dateFrom: Date;
    public dateTo: Date;
    public categoryId: number;
    public totalFrom: number;
    public totalTo: number;

    constructor() {
        this.userId = null;
        this.dateFrom = null;
        this.dateTo = null;
        this.categoryId = null;
        this.totalFrom = null;
        this.totalTo = null;
    }
}